### DISCUSSION TITLE
Are there any intentions for the Firebase Messaging alternative here?

### DISCUSSION THREAD

**Discussion by EgHubs** [supabase/supabase#3883] [Feature Requests]
Hope there are..

**Comment by Isaiah-Hamilton** [comment]
@EgHubs can you give me an example or explain better because I'm a little confused about what you're talking about. Then maybe I or someone from the Supabase team can help with your question.

**Reply by EgHubs** [reply]
in firebase there I can get the user token by a function and trigger an onTokenRefreash method and onBackgroundHandller in flutter. I can also send these messages from firebase functions, can I do that in Supabase?.
also is there a way to move my data from my current working released project from firebase to Supabase, especially user data auth.
How hard will it be?

**Reply by Isaiah-Hamilton** [reply]
> also is there a way to move my data from my current working released project from firebase to Supabase, especially user data auth.

@EgHubs There is a guide on how to transfer all your Firebase stuff into Supabase. Here is the [guide](https://github.com/supabase-community/firebase-to-supabase). That should help with your switching problems.
